homeassistant/components/homekit_controller/cover.py

```diff
@@ -265,30 +273,45 @@ async def async_set_cover_position(self, **kwargs: Any) -> None:
     async def async_set_cover_tilt_position(self, **kwargs: Any) -> None:
         """Move the cover tilt to a specific position."""
         tilt_position = kwargs[ATTR_TILT_POSITION]
+
+        def _get_min_max(
```

Comment:
Then I think there's a similar refactor in here where you set char based on the conditional and then do everything else based on whatever char is set to. Then you can drop both helpers. You would then have one call to put_characteristics and iirc fish the uuid out of char.type?

Reply:
Did some refactoring, but did not get the second part of your message.

Reply:
Got it I believe. Pushed another change for this one.